### PR TITLE
[MatrixCamp] add function tan

### PR DESCRIPTION
**Pull request by bxiiiiii**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #1968

**What this PR does / why we need it:**

implement Tan builtin function

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/builtin/unary/tan.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+				return resultVector, nil
+			},
+		},
+		{ // T_float64
+			Typ:        types.T_float64,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]float64)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
```
Comment (broccoliSpicy) [inline pkg/builtin/unary/tan.go:202]: perhaps we could reuse the original vector when possible

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]

File: pkg/builtin/unary/tan.go (hunk 1 above)
Comment (bxiiiiii) [reply pkg/builtin/unary/tan.go:207]: ok

**Review by broccoliSpicy — APPROVED** [review]
impeccable code...

**Comment by yingfeng** [comment]
@[bxiiiiii](https://github.com/bxiiiiii)

Pls re-submit this PR due to mis-corrected merge previously , sorry for the inconvenience!

**Pull request merged**